lib/rules/forbid-foreign-prop-types.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview Forbid using another component's propTypes
+ * @author Ian Christian Myers
+ */
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Constants
+// ------------------------------------------------------------------------------
+
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Forbid using another component\'s propTypes',
+      category: 'Best Practices',
+      recommended: false
+    }
+  },
+
+  create: function(context) {
+
+    return {
+      MemberExpression: function(node) {
+        if (node.property && node.property.type === 'Identifier' && node.property.name === 'propTypes' ||
+            node.property && node.property.type === 'Literal' && node.property.value === 'propTypes') {
+          context.report({
```

Comment:
it'd be helpful if the error could include the position of the warning, so that editors can jump the cursor right to it